Dockerfile

```diff
@@ -1,41 +1,53 @@
-# Build the Go binary in a separate stage utilizing Makefile
+# Build stage
 FROM golang:1.22 AS builder
 
-# Install necessary packages for the final image
+WORKDIR /app
+
+# Install Node.js and Yarn
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-    curl sudo gpg lsb-release software-properties-common \
+    curl gpg \
     && curl -fsSL https://deb.nodesource.com/setup_lts.x | bash - \
-    && curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor -o /usr/share/keyrings/yarn-archive-keyring.gpg \
-    && apt remove cmdtest \
-    && echo "deb [signed-by=/usr/share/keyrings/yarn-archive-keyring.gpg] https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
-    && apt-get update && apt-get install -y git yarn apt-utils
+    && apt-get install -y nodejs \
+    && npm install -g yarn
 
-WORKDIR /app
-COPY go.mod go.sum ./
-RUN go mod download
+# Copy the entire project
 COPY . .
 
+# Install contract dependencies
+RUN cd contracts && yarn install
+
+# Build the Go binary
 RUN make build
 
-# Use the official Ubuntu 22.04 image as a base for the final image
-FROM ubuntu:22.04 AS base
+# Final stage
+FROM ubuntu:22.04
+
+# Install necessary packages
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    nodejs npm \
+    && npm install -g yarn
+
+# Create the 'masa' user and set up the home directory
+RUN useradd -m -s /bin/bash masa && \
+    mkdir -p /home/masa/.masa && \
+    chown -R masa:masa /home/masa
 
+# Copy the built binary and set permissions
 COPY --from=builder /app/bin/masa-node /usr/bin/masa-node
 RUN chmod +x /usr/bin/masa-node
 
-# Create the 'masa' user and set up the home directory
-RUN useradd -m -s /bin/bash masa && mkdir -p /home/masa/.masa && chown -R masa:masa /home/masa
+# Copy contracts directory including node_modules
+COPY --from=builder --chown=masa:masa /app/contracts /home/masa/contracts
```

Comment:
contracts aren't read anymore from the filesystem, so this as well shouldn't be needed (the binary now doesn't ready contracts from the file location anymore)

https://github.com/masa-finance/masa-oracle/blob/609e2ecc7be7221bebb4be607392b84620fe7728/pkg/staking/abi.go#L30

```suggestion
```

Reply:
I think part of the issue that this PR is meant to address is this error when running `masa-node --faucet`:
```
ERRO[0000] [-] Failed to create staking client:[-] Failed to load contract addresses: open contracts/node_modules/@masa-finance/masa-token/addresses.json: no such file or directory 
ERRO[0000] [-] [-] Failed to load contract addresses: open contracts/node_modules/@masa-finance/masa-token/addresses.json: no such file or directory 
```

I haven't been able to dig into it very deeply but I _think_ it's because this code needs to load the embedded addresses.json file(s):
https://github.com/masa-finance/masa-oracle/blob/f09fb2076d4c1376ea855a12e72692c76ef32803/pkg/staking/config.go#L11

Maybe if that is fixed the Dockerfile won't need to copy over the node_modules?

Reply:
^ That's right, with the current setup, I'm not accessing addresses.json in the node_modules

@mudler I didn't realize the embed.go, but some research is telling me the node_modules won't be included in `//go:embed *`, maybe we should be more specific?

```
//go:embed node_modules/@masa-finance/masa-contracts-oracle/addresses.json
//go:embed node_modules/@masa-finance/masa-token/addresses.json
```